### ISSUE TITLE
Paypal module

### ISSUE THREAD

**Issue by pbright** [sonsandco/django-shoptools#17]

**Comment by pbright** [comment]
@gregplaysguitar How would you feel about me reimplementing this using https://github.com/paypal/PayPal-Python-SDK/tree/master/samples/payment?

Would mean adding a dependency on that and also its dependencies (libssl-dev and libffi-dev).

**Comment by gregplaysguitar** [comment]
Sounds fine - the dependencies are no issue so long as they're only required if the site uses Paypal - so they wouldn't go in setup.py

**Comment by pbright** [comment]
This has been done in Soohyang. 
https://github.com/sonsandco/soohyang/tree/master/ext/django-shoptools/shoptools/contrib/paypal

Did you want me to push it up here?

**Comment by gregplaysguitar** [comment]
Looks good - is this a drop-in replacement for django-dps? I think it should be a separate repo, if it's not actually related to shoptools at all. Ideally what we want is a documented interface for payment providers so we can swap them in and out easily rather than bundling everything with this repo.

django-paypal already exists though so we'd need to think of another name... How about django-paypaul

**Comment by pbright** [comment]
Haha, classic :D.

Yeah it's a replacement for django-dps. It is shoptools specific, there would be no point doing anything more general since there's already PayPal-Python-SDK. 

It's mostly just a wrapper around that, providing a make_payment method, transaction model and the other bits and pieces shoptools expects.

Really the only other thing it's doing is prevent Paypal from having anything to do with the addresses, for the reasons explained in the readme.

**Comment by gregplaysguitar** [comment]
How is it shoptools specific? Couldn't we use it in isolation for a non-shop payment -
a donation form or something like that?

**Comment by pbright** [comment]
I meant I couldn't see anyone using it outside of the shoptools, but you're right that there is a use case for it.

**Comment by gregplaysguitar** [comment]
Yeah, I can see it being useful. Do you want to make a repo for it?

**Comment by gregplaysguitar** [comment]
What we really need here is a reference for the payment interface - https://github.com/sonsandco/django-shoptools/blob/master/docs/reference/payment.md
